greenplumpython/pandas/dataframe.py

```diff
@@ -0,0 +1,235 @@
+        )[:]
+        return DataFrame(df)
+
+    def drop_duplicates(
+        self,
+        subset: Union[str, List[str], None] = None,
+        keep: Literal["first", "last", False] = "first",
+        inplace: bool = False,
+        ignore_index: bool = True,
+    ):
+        assert keep == "first", "Can only keep first occurrence"
+        assert inplace is False, "Cannot perform operation in place"
+        assert (
+            ignore_index is True
+        ), "DataFrame in GreenplumPython.pandas does not have an index column"
+        assert subset is not None  # FIXME: select distinct *
+        df = self._proxy.distinct_on(*subset)
+        return DataFrame(df)
+
+    def merge(
+        self,
+        right: "DataFrame",
+        how: Literal["left", "right", "outer", "inner", "cross"] = "inner",
+        on: Union[str, List[str]] = None,
+        left_on: str = None,
+        right_on: str = None,
+        left_index: bool = False,  # Not Used
+        right_index: bool = False,  # Not Used
+        sort: bool = False,
+        suffixes: str = "",
+        copy: bool = True,  # Not Used
+        indicator: bool = False,  # Not Used
+        validate: Optional[str] = None,  # Not Used
+    ):
+        how = "full" if how == "outer" else how
+        assert (
+            suffixes == ""
+        ), "Can't support yet automatically handle overlapping column's suffixes"
+        assert (
+            sort is False
+        ), "Can't support yet order result DataFrame lexicographically by the join key"
+        assert (
+            left_index is False and right_index is False
+        ), "DataFrame in GreenplumPython.pandas does not have an index column"
+        assert on is None, "Can't support duplicate columns name in both DataFrame"
+        df = self._proxy.join(right._proxy, how=how, cond=lambda s, o: s[left_on] == o[right_on])
+        return DataFrame(df)
+
+    def groupby(self, by: Union[str, List[str]]):
+        return DataFrameGroupBy(self._proxy.group_by(by))
+
+    def head(self, n: int):
+        df = self._proxy[:n]
+        return DataFrame(df)
+
+    def __repr__(self):
+        return self._proxy.__repr__()
+
+    def _repr_html_(self):
+        return self._proxy._repr_html_()  # type: ignore
+
+    def __iter__(self):
+        return self._proxy.__iter__()
+
+    def apply(self):
+        pass
+
+    def agg(self):
+        pass
+
+
+def read_sql(
+    sql: str,
+    con: engine,
+    index_col: Union[str, list[str]] = None,
+    coerce_float: bool = True,
+    params: Optional[List[str]] = None,
+    parse_dates: Optional[List[str]] = None,
+    columns: Optional[list[str]] = None,
+    chunksize: Optional[int] = None,
+):
+    database = db.Database(url=str(con.url))
+    try:
+        database.execute(f"SELECT * FROM {sql}")
+        df = database.create_dataframe(table_name=sql)
+        return DataFrame(df, con=database)
+    except:
+        return DataFrame(dataframe.DataFrame(query=sql, db=database))
```

Comment:
why try, except here?

Reply:
To see if sql it is a table name or a query